src/plugins/data/public/data_sources/datasource/factory.test.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DataSourceFactory } from './factory';
+import { DataSource } from './datasource';
+import { IndexPattern } from '../../index_patterns';
+
+class MockDataSource extends DataSource<any, any, any, any, any> {
+  private readonly indexPatterns;
+
+  constructor({
+    name,
+    type,
+    metadata,
+    indexPatterns,
+  }: {
+    name: string;
+    type: string;
+    metadata: any;
+    indexPatterns: IndexPattern;
+  }) {
+    super(name, type, metadata);
+    this.indexPatterns = indexPatterns;
+  }
+
+  async getDataSet(dataSetParams?: any) {
+    await this.indexPatterns.ensureDefaultIndexPattern();
+    return await this.indexPatterns.getCache();
+  }
+
+  async testConnection(): Promise<void> {
+    throw new Error('This operation is not supported for this class.');
+  }
+
+  async runQuery(queryParams: any) {
+    return null;
```

Comment:
Does this have to be `null`? Such things are usually returned as `undefined` because while serializing, a prop with a value of `undefined` is omitted.